src/auth/index.js

```diff
@@ -231,41 +265,41 @@ export default class Auth {
     }
 
     let newUserAuth = {};
+    let setters = [];
 
-    return (authType ?
-      this.storage.set(authCommon.USER_LOGGED_IN_PT_KEY, authType) :
-      Promise.resolve()
-    ).then(() => {
-      if (json[this.codec.refreshToken]) {
-        let rt = json[this.codec.refreshToken];
-        delete json[this.codec.refreshToken];
-        return this.storage.set(authCommon.REFRESH_TOKEN_KEY, rt);
-      }
-    }).then(() => {
-      if (json[this.codec.deviceId]) {
-        const deviceId = json[this.codec.deviceId];
-        delete json[this.codec.deviceId];
-        return this.storage.set(authCommon.DEVICE_ID_KEY, deviceId);
-      }
-      return;
-    }).then(() => {
-      // Merge in new fields with old fields. Typically the first json value
-      // is complete with every field inside a user auth, but subsequent requests
-      // do not include everything. This merging behavior is safe so long as json
-      // value responses with absent fields do not indicate that the field should
-      // be unset.
-      if (json[this.codec.accessToken]) {
-        newUserAuth.accessToken = json[this.codec.accessToken];
-      }
-      if (json[this.codec.userId]) {
-        newUserAuth.userId = json[this.codec.userId];
-      }
+    if (authType) {
+      this.loggedInProviderType = authType;
+      setters.push(this.storage.set(authCommon.USER_LOGGED_IN_PT_KEY, authType));
+    }
 
-      return this._get();
-    }).then((auth) => {
-      newUserAuth = Object.assign(auth, newUserAuth);
-      return this.storage.set(authCommon.USER_AUTH_KEY, JSON.stringify(newUserAuth));
-    });
+    if (json[this.codec.refreshToken]) {
+      this.rt = json[this.codec.refreshToken];
+      delete json[this.codec.refreshToken];
+      setters.push(this.storage.set(authCommon.REFRESH_TOKEN_KEY, this.rt));
+    }
+
+    if (json[this.codec.deviceId]) {
+      this.deviceId = json[this.codec.deviceId];
+      delete json[this.codec.deviceId];
+      setters.push(this.storage.set(authCommon.DEVICE_ID_KEY, this.deviceId));
+    }
+
+    // Merge in new fields with old fields. Typically the first json value
+    // is complete with every field inside a user auth, but subsequent requests
+    // do not include everything. This merging behavior is safe so long as json
+    // value responses with absent fields do not indicate that the field should
+    // be unset.
+    if (json[this.codec.accessToken]) {
+      newUserAuth.accessToken = json[this.codec.accessToken];
+    }
+    if (json[this.codec.userId]) {
+      newUserAuth.userId = json[this.codec.userId];
+    }
+
+    this.auth = Object.assign(!!this.auth && {}, newUserAuth);
```

Comment:
[nit] reduce the magic with `this.auth ? this.auth : {}`

Reply:
Actually, I'm pretty sure the code as is, has a bug and loses the past auth information rendering this.auth to always be set to newUserAuth, not merged. This then causes `this.authedId = this.auth.userId;` to be set to undefined on a token refresh and then permanently lost.

Reply:
👍